src/h5web/vis-packs/core/utils.ts

```diff
@@ -304,7 +304,7 @@ export function getBaseArray<T>(
   value: T[] | undefined,
   rawDims: number[]
 ): ndarray<T> | undefined {
-  return value && ndarray<T>(value.flat(Infinity) as T[], rawDims);
+  return value && ndarray<T>(value.flat(rawDims.length - 1) as T[], rawDims);
```

Comment:
This ensures that complex data does not get flattened all the way: complex arrays are `ndarrays` that contain two-value arrays as value.

It does not change anything for other types of arrays.

Reply:
Can you please explain why this doesn't change anything for other types of arrays?

Reply:
Imagine a 2D array of integers:
```
[
  [0, 2],
  [1, 9]
]
```

Converting this array to 1D require a `.flat(1)` i.e. `.flat(nDims - 1)`. Yielding the same as `.flat(Infinity)`.

Reply:
Ah right because `rawDims` comes from the HDF5 type. Makes sense 👍

Reply:
Maybe just add a comment to that regard somewhere?